x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/Querier.java

```diff
@@ -225,8 +226,13 @@ public void onResponse(Page page) {
                 }
             }
 
-            // 1. consume all pages received
-            consumeRowSet(page.rowSet());
+            try {
```

Comment:
Since we already have access to the listener, how about returning a bool instead of throwing an exception.
There's nothing wrong with it yet consumeRowSet is used only internally so promoting the condition to an exception only to catch it later  looks a bit wasteful to me.

suggestion:
```
if (consumeRowSet(page.rowSet()) == false) {
   return;
}
```